tooling/ast_fuzzer/src/program/func.rs

```diff
@@ -729,4 +733,83 @@ impl<'a> FunctionContext<'a> {
 
         Ok(expr)
     }
+
+    /// Generate a `loop` loop.
+    fn gen_loop(&mut self, u: &mut Unstructured) -> arbitrary::Result<Expression> {
+        // Declare break index variable visible in the loop body. Do not include it
+        // in the locals the generator would be able to manipulate, as it could
+        // lead to the loop becoming infinite.
+        let idx_type = types::U32;
+        let idx_id = self.next_local_id();
+        let idx_name = format!("idx_{}", make_name(idx_id.0 as usize, false));
+        let idx_ident =
+            expr::ident_inner(VariableId::Local(idx_id), true, idx_name.clone(), idx_type);
+
+        // Decrease budget so we don't nest endlessly.
+        self.decrease_budget(1);
+
+        // Start building the loop harness, initialize index to 0
+        let mut stmts = vec![Expression::Let(Let {
+            id: idx_id,
+            mutable: true,
+            name: idx_name,
+            expression: Box::new(expr::u32_literal(0)),
+        })];
+
+        // Get the randomized loop body
+        let inner_block = self.gen_block(u, &Type::Unit)?;
+        let Expression::Block(mut inner_stmts) = inner_block else {
+            unreachable!("generated a non-block expression: {}", inner_block)
```

Comment:
Technically `gen_block` is currently allowed to return a non-block expression if it's called when there is no budget, which could be the case after decreasing it on line #749.

Reply:
I would suggest not unwrapping here, but instead just adding it to a sub block as it is:
```
Block(vec![
  Let(idx, 0), 
  Loop(
    Block(vec![      
      If(
        Binary(idx, Eq, max_size),
        Break,
        Block(vec![
          Assign(idx, Binary(idx, Add, 1)),
          inner_block
        ])
      )
    ])
  )
])
```